verilog/tools/kythe/indexing_facts_tree_extractor.cc

```diff
@@ -255,6 +258,21 @@ void IndexingFactsTreeExtractor::Visit(const SyntaxTreeNode& node) {
       ExtractTypeDeclaration(node);
       break;
     }
+    case NodeEnum::kLoopGenerateConstruct:
```

Comment:
Do we need `kConditionalGenerateConstruct`, `kGenerateRegion`?

Reply:
I don know where these tags appear? can you point me.

Reply:
https://cs.opensource.google/verible/verible/+/master:verilog/CST/verilog_nonterminals.h

e.g. `module m; if (condition) ...` and `module m; for(genvar i=0; ...) ...`

Reply:
I think we don't we have already test cases for these two examples and they are working fine.

Reply:
The following works already?

```systemverilog
module m #(int N) ();
  for (genvar i=0; i<N; ++i) begin
    wire foo, bar;
    assign foo = bar;
  end
  for (genvar i=0; i<N; ++i) begin
    wire foo, bar;
    assign foo = bar;
  end
endmodule
```

The loops are duplicate, but aside from N, all other symbols should be distinct from the other loop.

BTW, the relevant passage on scoping from the SV-LRM is chapter 27.  For loops 27.4, and for conditional 27.5.

Can you help me with a pointer?  I haven't mentally cached our corpus of test cases.

If only I had a tool that could search files by language-feature (CST-based)...

Reply:
I added some comments below for the files with these examples.

Reply:
Thanks for the comments.